registry/registry.go

```diff
@@ -99,7 +101,12 @@ func doRequest(req *http.Request, jar http.CookieJar, timeout TimeoutType, secur
 			return false
 		}
 
-		hostDir := path.Join("/etc/docker/certs.d", req.URL.Host)
+		if runtime.GOOS == "windows" {
+			hostDir = filepath.Join(os.TempDir(), "/docker/certs.d")
```

Comment:
Yup, looked at this library. The whole "where do we store various settings" debate is one I haven't tackled yet, and isn't straightforward. Hence I just mirrored what Linux does and used the file system. Re-evaluating where Windows stores persistent state is something which needs a higher level view on, particularly if the daemon moves to a Windows service. This is something I have on my list for future improvements once we have the daemon at a running state with the current set of PRs. For now, keeping a file seems to make sense.